[ADP-3486] Add a simple bash script to generate random deposit payments

abailly opened the pull request:
See the script's usage for details. This PR is in draft because I think the script is wrong: The wallet is not able to detect payment transactions it's making.

abailly committed:
Add a simple bash script to generate random deposit payments

See the script's usage for details

Anviking commented on scripts/tx-gen/gen-deposits.sh:
```diff
@@ -0,0 +1,119 @@
+  acct_id=$((r % 10000 ))
+
+  # derive public address
+  account_address=$(cat root.prv | cardano-address key child "1852H/1815H/${acct_id}H/0/0" | cardano-address key public --with-chain-code | cardano-address address payment --network-tag preprod)
```
I believe it should be

```suggestion
  account_address=$(cat root.prv | cardano-address key child "1857H/1815H/0H/0/${addr_id}" | cardano-address key public --with-chain-code | cardano-address address payment --network-tag preprod)
```

based off 

```
deriveAddress s ix = rootXPrv s / 1857' / 1815' / 0' / 0 / ix
```

from https://github.com/cardano-foundation/cardano-wallet-agda/blob/43d1f83e4af85d318dcdbd912c8620bd05b60f78/lib/customer-deposit-wallet-pure/agda/Specification.lagda.md#address-derivation

abailly committed:
Generate payments up to ₳1000 for 10 accounts

abailly committed:
Use last index for derivnation instead of 3rd

abailly committed:
Better handling of limit conditions

Also add some documentation showing output of the script.

abailly committed:
Properly quote here document delimiter to escape content

Without the quotes, the body of the here document is interpreted by
the shell potentially leading to all kind of weird behaviour.

abailly committed:
Fix comment

abailly committed:
Compute tx number from pre-existing tx.X.signed file

Also output the total remaining UTxO value from accumulated deposit
payments. These 2 features together are useful to create chained
transactions without having to wait for one to be submitted.

abailly committed:
Allow bounding maximum deposited value to less than total utxo

This makes it possible to use a large UTxO in smaller chunks

abailly committed:
Change base derivation to match deposit wallet's specification

abailly committed:
Reduce amounts "deposited"

abailly committed:
Fix shellcheck warnings

abailly committed:
Submit deposit transactions at a random interval

abailly committed:
Address more shellcheck warnings

abailly committed:
Bump version of bump

paolino reviewed (approved):
Thanks
